server/src/test/java/com/linecorp/centraldogma/server/internal/replication/ZooKeeperCommandExecutorTest.java

```diff
@@ -254,13 +255,55 @@ void stopWhileWaitingForInitialQuorum() throws Exception {
                                .hasMessageContaining("before joining");
     }
 
+    @Test
+    void hierarchicalQuorums() throws Throwable {
```

Comment:
Question:
Can we check how it works when the weight is different? for example:
```
(x) means the weight of the replica

group.1=1(2):2(1):3(1) (and what happens if 1 is down?)
or
group.1=1(1):2(0):3(0)
or 
group.1=1(0):2(0):3(0)

with
group.2=4:5:6
group.3=7:8:9
```

Reply:
> Can we check how it works when the weight is different?

I will add test cases for that.

> group.1=1(2):2(1):3(1) (and what happens if 1 is down?)

(Maybe) In this case, Group 1 is excluded from quorums, however, the majority is still satisfied because the remaining two groups can vote.